services/api/events.js

```diff
@@ -20,43 +20,92 @@ const eventSchema = Schema({
 
 const eventModel = conn.model('events', eventSchema)
 
+const ajv = new Ajv();
+const eventQuerySchema = {
+    type: 'object',
+    properties: {
+        timestamp: {
+            type: 'object',
+            properties: {
+                start: {type: 'integer'},
+                end: {type: 'integer'}
+            },
+            additionalProperties: false
+        },
+        events: {
+            type: 'object',
+            properties: {
+                gt: {type: 'string'},
+                lt: {type: 'string'},
+                gte: {type: 'string'},
+                lte: {type: 'string'}
+            },
+            minProperties: 1,
+            additionalProperties: false
+        },
+        analyzers: {
+            type: 'array',
+            items: {
+                type: 'string'
+            },
+        },
+        types: {
+            type: 'array',
+            items: {
+                type: 'string'
+            },
+        }
+    }
+    ,additionalProperties: false
+}
+
+const eventQueryValidator = ajv.compile(eventQuerySchema);
+
+function validateEventQuery(req, res, next) {
+    if(!eventQueryValidator(req.body)) {
+        return next(createError(400, 'Bad Request'));
```

Comment:
It would be better to return (or at least to log) the error cause so that the API caller knows what's going wrong. This could be done by accessing eventQueryValidator.errors (according to the README.md of [ajv](https://github.com/epoberezkin/ajv)).